src/status_im/contexts/profile/edit/header/events_test.cljs

```diff
@@ -0,0 +1,26 @@
+(ns status-im.contexts.profile.edit.header.events-test
+  (:require [cljs.test :refer [deftest is]]
+            matcher-combinators.test
+            [status-im.common.profile-picture-picker.view :as photo-picker]
+            [status-im.contexts.profile.edit.header.events :as sut]))
+
+(deftest edit-picture-test
+  (let [picture  "new-picture"
+        key-uid  "key-uid"
+        cofx     {:db {:profile/profile {:key-uid key-uid}}}
+        expected {:json-rpc/call
+                  [{:method     "multiaccounts_storeIdentityImage"
+                    :params     [key-uid picture 0 0 photo-picker/crop-size photo-picker/crop-size]
+                    :on-success fn?}]}]
```

Comment:
For knowledge sake, not necessarily a problem in this PR :)

Sometimes the `on-success` part of the effect `json-rpc/call` is not straightforward and we might want to test it. In those cases I prefer to dispatch to a new event, because then that event can be tested manually (via REPL) or automatically (unit tests).

When the logic is behind the anonymous function passed to `:on-success`, it becomes a black box.

Example:

```clojure
(defn edit-profile-picture
  [{:keys [db]} [picture crop-width crop-height]]
  (let [key-uid     (get-in db [:profile/profile :key-uid])
        crop-width  (or crop-width photo-picker/crop-size)
        crop-height (or crop-height photo-picker/crop-size)]
    {:fx [[:json-rpc/call
           [{:method     "multiaccounts_storeIdentityImage"
             :params     [key-uid picture 0 0 crop-width crop-height]
             :on-success [:profile/update-local-picture-success]}]]]}))

(rf/reg-event-fx :profile/edit-picture edit-profile-picture)

(rf/reg-event-fx :profile/edit-profile-picture-success
 (fn [_ [images]]
   {:fx [[:dispatch [:profile/update-local-picture (reverse images)]]
         [:dispatch [:toasts/upsert
                     {:type  :positive
                      :theme :dark
                      :text  (i18n/label :t/profile-picture-added)}]]]}))

```

Reply:
thanks, applied the suggestion